src/components/ReleaseMonitor/ReleaseFilters.tsx

```diff
@@ -0,0 +1,221 @@
+        );
+        setIsAttributeMenuOpen(!isAttributeMenuOpen);
+      }}
+    >
+      <MenuContent>
+        <MenuList>
+          <MenuItem itemId="Application">Application</MenuItem>
+          <MenuItem itemId="Namespace">Namespace</MenuItem>
+          <MenuItem itemId="Release Plan">Release Plan</MenuItem>
+          <MenuItem itemId="Status">Status</MenuItem>
+        </MenuList>
+      </MenuContent>
+    </Menu>
+  );
+
+  const attributeDropdown = (
+    <div ref={attributeContainerRef}>
+      <Popper
+        trigger={attributeToggle}
+        triggerRef={attributeToggleRef}
+        popper={attributeMenu}
+        popperRef={attributeMenuRef}
+        appendTo={attributeContainerRef.current || undefined}
+        isVisible={isAttributeMenuOpen}
+      />
+    </div>
+  );
+
+  const renderActiveFilter = () => {
+    switch (activeAttributeMenu) {
+      case 'Application':
+        return (
+          <ToolbarFilter
+            chips={applicationSelections}
+            deleteChip={(_, chip) =>
+              onApplicationSelectionsChange(applicationSelections.filter((s) => s !== chip))
+            }
+            deleteChipGroup={() => onApplicationSelectionsChange([])}
+            categoryName="application"
+          >
+            <ApplicationFilter
+              applications={applications}
+              onSelectionsChange={onApplicationSelectionsChange}
+            />
+          </ToolbarFilter>
+        );
+      case 'Namespace':
+        return (
+          <ToolbarFilter
+            chips={tenantSelections}
+            deleteChip={() => onTenantSelect(undefined)}
+            deleteChipGroup={() => onTenantSelect(undefined)}
+            categoryName="tenant"
+          >
+            <TenantFilter
+              tenants={tenants}
+              selections={tenantSelections}
+              onSelect={onTenantSelect}
+            />
+          </ToolbarFilter>
+        );
+      case 'Release Plan':
+        return (
+          <ToolbarFilter
+            chips={searchValue !== '' ? [searchValue] : ([] as string[])}
+            deleteChip={() => onSearchChange('')}
+            deleteChipGroup={() => onSearchChange('')}
+            categoryName="releaseplan"
+          >
+            <SearchFilter value={searchValue} onChange={onSearchChange} />
+          </ToolbarFilter>
+        );
+      case 'Status':
+        return (
+          <ToolbarFilter
+            chips={statusSelection !== '' ? [statusSelection] : ([] as string[])}
+            deleteChip={() => onStatusSelect(undefined)}
+            deleteChipGroup={() => onStatusSelect(undefined)}
+            categoryName="status"
+          >
+            <StatusFilter selection={statusSelection} onSelect={onStatusSelect} />
+          </ToolbarFilter>
+        );
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <Toolbar id="attribute-search-filter-toolbar" clearAllFilters={onClearFilters}>
```

Comment:
maybe we could try to use `BaseTextFilterToolbar` component? it should simplify things and avoid repeating code as we have in the components created in this PR (e.g. `ApplicationFilter`, `StatusFilter`, ...) :) - if you search for this component usage in our codebase, you'll see that we're using it in many different places: 

![Screenshot 2025-07-01 at 14 12 55](https://github.com/user-attachments/assets/893ec90c-2162-4048-a539-5fc71195f170)

Reply:
Seems it's different than what I wrote, the new "release monitor" page add all conditions together.
It's better to create another JIRA issue to track this change as I think it's not easy replacement. What's your thinking?

Reply:
sure, if the other team members are also fine with it :) 

could you please create a ticket to do this refactor? basically, replace the `ReleaseFilters` component (altogether with `ApplicationFilter`, `SearchFilter`, `StatusFilter`, `TenantFilter`) with `BaseTextFilterToolbar`.  you can use `src/components/Filter/toolbars/PipelineRunsFilterToolbar.tsx` as inspiration :)

Reply:
File the [ROK-674](https://issues.redhat.com/browse/ROK-674)